src/traits-lowering-rules.md

```diff
@@ -270,25 +300,26 @@ Given an impl that contains:
 
 ```rust,ignore
 impl<P0..Pn> Trait<A1..An> for A0
-where WC
+where WC_impl
 {
-    type AssocType<Pn+1..Pm> where WC1 = T;
+    type AssocType<Pn+1..Pm> = T;
 }
 ```
 
-We produce the following rule:
+and our where clause `WC1` on the trait associated type from above, we
+produce the following rule:
 
 ```text
 // Rule Normalize-From-Impl
 forall<P0..Pm> {
   forall<Pn+1..Pm> {
     Normalize(<A0 as Trait<A1..An>>::AssocType<Pn+1..Pm> -> T) :-
-      WC && WC1
+      WC_impl && WC1
```

Comment:
`WC_impl` can actually be replaced with `Implemented(A0: Trait)` (and that’s what we do in chalk)

Reply:
Done. Do you think it's more or less confusing to keep `WC_impl` around then? :)

Reply:
Let’s keep it, so that it looks like a general enough impl.

Reply:
Sounds good. I added a note about `WC_impl` being unused just so it doesn't seem like a mistake.